test/fast/Unit tests/nvm_find_project_dir

```diff
@@ -31,3 +31,6 @@ ACTUAL="$(PWD=$TEST_DIR/no-nesting-n_m nvm_find_project_dir)"
 
 ACTUAL="$(PWD=$TEST_DIR/no-nesting-pkg nvm_find_project_dir)"
 [ "${ACTUAL}" = "$TEST_DIR/no-nesting-pkg" ] || die "no-nesting-pkg: got ${ACTUAL}"
+
+ACTUAL="$(PWD="." nvm_find_project_dir)"
+[ "${ACTUAL}" = "." ] || die "insufficient permissions for pwd: got ${ACTUAL}"
```

Comment:
this is good for a unit test, but it'd be ideal to actually reproduce the failure with `su`, perhaps in a "slow" test

Reply:
I don't think using `su` is a good idea, since the test would need to create a user, and need `sudo`, and would be too OS-dependant (at least for local testing).

It seems to be possible to reproduce the same scenario when the current directory is deleted, but I think it also may be shell- and OS-specific.

Reply:
If there's only specific shells or OS's where this can happen, that needs to be documented in a comment or something at least.

Reply:
I believe the original bug can affect multiple different shells and OSes. I personally encountered it on macOS using zsh when using `su`.

I don't think testing this specific case is feasible as it requires `sudo` to set up a user, and may be more effort than it's worth having to log in as that user.

My suggestion to use deleting the current directory came from [this SE post](https://unix.stackexchange.com/q/606465) but this seems to be specific to a certain combination of an OS and a shell, as in general a unix directory wouldn't be actually deleted until the link count becomes 0, which wouldn't happen until the shell process, which has the directory opened, terminates.

And I can confirm that at least on macOS in zsh, deleting a directory does _not_ change `$PWD` (or the output of the `pwd` command)

Ultimately, I can't think of a good way to integration-test this unfortunately. And even if this behaviour has some unintended consequences, they cannot be worse than the script getting into an infinite loop.

Reply:
Fair enough, thanks for the analysis.